src/mutex.c

```diff
@@ -69,14 +69,21 @@ void
 malloc_mutex_lock_slow(malloc_mutex_t *mutex) {
 	mutex_prof_data_t *data = &mutex->prof_data;
 
-	{//TODO: a smart spin policy
+	if (ncpus == 1) {
+		goto label_spin_done;
+	}
+
+	int cnt = 0, max_cnt = MALLOC_MUTEX_MAX_SPIN;
+	do {
+		CPU_SPINWAIT;
```

Comment:
Why isn't this using spinner?  The problem with busy spinning like this is that it beats hard on the cache coherency hardware, whereas exponential backoff does not.